main/src/library/CommutativeSemiGroup.flix

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright 2022 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+///
+/// A type class for types that form a commutative semigroup.
+///
+pub class CommutativeSemiGroup[a] with SemiGroup[a] {
+
+    ///
+    /// An associative binary operation on `a`.
+    ///
+    pub def combine(x: a, y: a): a = SemiGroup.combine(x, y)
+
+    ///
+    /// Returns `x` combined with itself `n` times.
+    ///
+    pub def combineN(x: a, n: Int32): a = SemiGroup.combineN(x, n)
+
+    law commutative: forall(x: a, y: a) with Eq[a] . CommutativeSemiGroup.combine(x, y) == CommutativeCommutativeSemiGroup.combine(y, x)
+
+}
+
+instance CommutativeSemiGroup[Unit] {
+}
+
+instance CommutativeSemiGroup[Option[a]] with CommutativeSemiGroup[a] {
```

Comment:
- Instances for enums (e.g. Option) belong in their namespace files. (This should also be changed for plain SemiGroup).
- The {} braces can be dropped.